estimator/estimator.py

```diff
@@ -53,57 +69,142 @@ def __init__(self, epochs=500, batch_size=128, learn_rate=0.001, keep_prob=0.4):
+            aggregate_results(cpu_estimated, "tw", config_num, model_name)
+        elif config_num==3:
+            cpu_estimated = estimate_job_time_time_writter(times, 0.014, model_name)
+            aggregate_results(cpu_estimated, "tw", config_num, model_name)
+        else:
+            raise Exception("Invalid config: ", config)
+    return cpu_estimated
+
+def estimate_job_time_linreg(times, a, b, model_name):
+	
+	###################################
+	#  Linear Regression Estimation   # 
+	###################################
+    total_iterations = 0
+    if model_name == "lenet":
+        total_iterations = 30000
+    else:
+        total_iterations = 500
+
+	# Using a and b proportions, find corresponding epoch number
+    a_epochs = round(total_iterations * a)
+    times_iters = times[1:-1]
+    b_epochs = len(times_iters)
+
+    a_epochs_times = 0
+    b_epochs_times = 0
+    for i in times_iters[:a_epochs]:
+        a_epochs_times += i
+    for j in times_iters:
+        b_epochs_times += j
+    a_epochs_times += times[0]
+    
+    b_epochs_times += times[-1]
+    b_epochs_times += times[0]
+    
+    cpu_estimated = float('inf')
+    
+    slope = (b_epochs_times - a_epochs_times) / (b_epochs - a_epochs)
+    b = a_epochs_times - (slope * a_epochs)
+    
+    cpu_estimated = (slope*total_iterations)+b
+    return cpu_estimated
+
+def estimate_job_time_time_writter(times, proportion, model_name):
 	###################################
-	#####    ESTIMATE JOB TIME    #####
+	#####    Detailed Estimator   #####
 	###################################
-	cpu_estimated = float('inf')
-	gpu_estimated = float('inf')
-
-	job.epochs = original_epochs
-	job.cpu_compl_time = cpu_estimated
-	job.gpu_compl_time = gpu_estimated
-	job.cpu_err = 0.1
-	job.gpu_err = 0.1
-
-# job: the job whose CPU and GPU time should be estimated
-# num_epochs: the number of epochs to run the estimation job for
-def estimate_job_time_time_writter(job, num_epochs):
-	original_epochs = job.epochs
-	job.epochs = num_epochs
-
-	times = []
-	for _ in range(self.num_samples):
-		output, error = jopCopy.run_cpu()
-		measurements = time_writter.parse_output(output)
-		times.append(sum(measurements))
-		
-	avgTime = sum(times) / len(times)
-	cpu_estimated = avgTime / a * original_epochs
-
-	times = []
-	for _ in range(self.num_samples):
-		output, error = jopCopy.run_gpu()
-		measurements = time_writter.parse_output(output)
-		times.append(sum(measurements))
-		
-	avgTime = sum(times) / len(times)
-	gpu_estimated = avgTime / a * original_epochs
-
-	job.epochs = original_epochs
-	job.cpu_compl_time = cpu_estimated
-	job.gpu_compl_time = gpu_estimated
-	job.cpu_err = 0.1
-	job.gpu_err = 0.1
\ No newline at end of file
+    total_iterations = 0
+    if model_name == "lenet":
+        total_iterations = 30000
+    else:
+        total_iterations = 500
+    cpu_estimated = float('inf')
+    
+    sum_times = 0
+    for i in times[1:-1]:
+        sum_times += i

```

Comment:
this should be sum_times += times[i], or you could get rid of the entire loop with sum(times[1:-1])